Tasks/Gradle/CodeAnalysis/SonarQube/metrics.ts

```diff
@@ -72,14 +141,19 @@ export class SonarQubeMetrics {
 
         tl.debug(`[SQ] Getting the quality gate status `);
 
+        if (SonarQubeMetrics.isCached(this.analysisDetails)) {
+            return Q.when(SonarQubeMetrics.getQualityGateStatus(this.analysisDetails));
+        }
+
         return this.getAnalysisId()
             .then((analysisId:string) => {
                 tl.debug(`[SQ] Analysis ID: ${analysisId}`);
-                var status =  this.getAnalysisStatus(analysisId);
-                tl.debug(`[SQ] Analysis status: ${status}`);
```

Comment:
This is an unrelated change: while this looks like it is printing the analysis status, in fact this is a promise that has not yet been resolved. The promise framework simply unwraps the promise before returning it - therefore, attempting to print here simple results in _[Promise promise]_.

Fixed by resolving the promise before printing it.